lib/index.js

```diff
@@ -20,18 +27,32 @@ const internals = {
       domain: 'localhost',
       ignoreErrors: true,
       clearInvalid: true
-    },
-    sso: {}
+    }
   }
 };
 
 exports.register = function (plugin, options) {
   const settings = Hoek.applyToDefaults(internals.defaults, options);
 
+  Hoek.assert(settings.baseUrl, 'baseUrl is required');
+  Hoek.assert(settings.ssoUrl, 'ssoUrl is required');
+  Hoek.assert(settings.apiBaseUrl, 'apiBaseUrl is required');
+  Hoek.assert(settings.keyId, 'keyId is required');
+  Hoek.assert(settings.keyPath, 'keyId is required');
+  Hoek.assert(typeof settings.permissions === 'object', 'permissions must be an object');
```

Comment:
This should check for `null` too.

Reply:
I did have that, but with the Hoek.defaults it won't be possible to be null